docs/Account.md

```diff
@@ -0,0 +1,331 @@
+
+To use Signer, pass a private key when instantiating the class:
+
+```python
+from utils.Signer import Signer
+
+signer = Signer(123456789987654321)
+```
+
+Then send transactions with `send_transaction` method.
+
+```python
+await signer.send_transaction(account, contract_address, 'method_name', [])
+```
+
+
+## Message format
+
+The idea is for all user intent to be encoded into a `Message` as follows:
+
+```c#
+struct Message:
+    member sender: felt
+    member to: felt
+    member selector: felt
+    member calldata: felt*
+    member calldata_size: felt
+    member nonce: felt
+end
+```
+
+Where:
+
+- `sender` is the Account contract address. It is included to prevent transaction replays in case there's another Account contract controlled by the same public keys.
+- `to` address of the target contract of the message
+- `selector` is the selector of the function to be called on the target contract
+- `calldata` is an array representing the function parameters
+- `nonce` is an unique identifier of this message to prevent transaction replays
+
+This message is consumed by the `execute` method, which acts as a single entrypoint for all user interaction with any contract, including managing the account contract itself. That's why if you want to change the public key controlling the Account, you would send a transaction targeting the very Account contract:
+
+```python
+await signer.send_transaction(account, account.contract_address, 'set_public_key', [NEW_KEY])
+```
+
+Notice that Signer's `send_transaction` calls `execute` under the hood.
+
+Or if you want to update the Account's L1 address on the `AccountRegistry` contract, you would 
+
+```python
+await signer.send_transaction(account, registry.contract_address, 'set_L1_address', [NEW_ADDRESS])
+```
+
+You can read more about how messages are structured and hashed in the [Account message scheme  discussion](https://github.com/OpenZeppelin/cairo-contracts/discussions/24).
+
+
+## API Specification
+
+This in a nutshell is the Account contract public API:
+
+```cairo
+func get_public_key() -> (res: felt)
+func get_address() -> (res: felt)
+func get_nonce() -> (res: felt)
+
+func set_public_key(new_public_key: felt)
+func initialize(_address: felt)
+
+func is_valid_signature(hash: felt,
+        signature_len: felt,
+        signature: felt*
+    )
+
+func execute(
+        to: felt,
+        selector: felt,
+        calldata_len: felt,
+        calldata: felt*,
+        nonce: felt
+    ) -> (response : felt):
+```
+
+> Notice we're omitting `initialize` from the API spec since we consider it a temporary artifact of the lack of `this.address` and we hope to remove it as soon as possible.
+
+#### `get_public_key`
+
+Returns the public key associated with the Account contract.
+
+##### Parameters:
+
+```
+None
+```
+
+##### Returns:
+```
+(implicit) syscall_ptr : felt*
+(implicit) pedersen_ptr: HashBuiltin*
+(implicit) range_check_ptr
```

Comment:
Would it be bad to remove these from the docs? Do they mean anything relevant to the person reading?

If not, can we at least switch the order so the actual return values are featured first?